cmake/utils.cmake

```diff
@@ -53,12 +53,12 @@ endfunction()
 # )
 #
 # Defines a new test executable target with the given target name, sources, and
-# dependencies.  Implicitly adds DEPENDS on GTest::GTest and GTest::Main.
+# dependencies.  Implicitly adds DEPENDS on GTest::Main.
 function(cc_test name)
   set(multi DEPENDS SOURCES)
   cmake_parse_arguments(cct "" "" "${multi}" ${ARGN})
 
-  list(APPEND cct_DEPENDS GTest::GTest GTest::Main)
+  list(APPEND cct_DEPENDS GTest::Main)
```

Comment:
There's no harm that I'm aware of, and I haven't benchmarked it, so I have no reason to suspect this is appreciably faster/smaller/whatever.

The only reason I proposed this PR is that I noticed in gtest's docs that they suggest just using one or the other rather than both. eg search for gtest_main on this page:

https://github.com/google/googletest/tree/master/googletest/README.md

Also here:
https://github.com/google/googletest/blob/master/googletest/CMakeLists.txt#L92

If we're happy with this as is, I'm happy to abandon this PR.

Reply:
I can't say I care too much either way so if you think this is better/less confusing/more in line with recommendations I'm okay with it too.

Basically: up to you.

Reply:
It could be confusing to have different standards in cmake than we do in blaze and CocoaPods where it is policy to not rely on transitive dependencies.

Reply:
Yeah, that's essentially the concern I was trying to raise, just done more succinctly. Thanks, @paulb777.